__main__.py

```diff
@@ -35,5 +38,16 @@ def build() -> None:
     runpy.run_module('coherent.build', run_name='__main__')
 
 
+@app.command(context_settings=dict(allow_extra_args=True))
+def tag(
+    tag_name: str,
+    context: typer.Context,
+    location: Annotated[str, typer.Option('-C', help='Path to repository.')] = '.',
+) -> None:
+    if tag_name in Versioned.semantic_increment:
+        tag_name = semver(repo(location).get_next_version(tag_name))
```

Comment:
I note that the call to `semver()` doesn't happen if a direct tag is passed. That means a user can pass a non-semver tag. I think we want to disallow that. Let's call `semver()` unconditionally.

Reply:
It is worth noting then that we're forcing semver format. Or maybe add an option to disable it somewhere, but that's overcomplicated--I would opt for deferring that until someone finds it necessary (why would they use this utility then?).